R/assocTestSingle.R

```diff
@@ -93,10 +106,25 @@ setMethod("assocTestSingle",
 setMethod("assocTestSingle",
           "GenotypeIterator",
           function(gdsobj, null.model, test=c("Score", "Score.SPA", "BinomiRare", "CMP"),
-                   recalc.pval.thresh=0.05, GxE=NULL,
+                   recalc.pval.thresh=0.05, approx.score.SE=FALSE, GxE=NULL,
                    male.diploid=TRUE, verbose=TRUE) {
               test <- match.arg(test)
 
+              # Convert old null model format if necessary.
+              null.model <- .updateNullModelFormat(null.model)
+
+              # check null.model for needed components
+              calc.score <- test %in% c("Score", "Score.SPA") | (recalc.pval.thresh < 1)
+              if(calc.score && approx.score.SE && is.null(null.model$score.se.ratio)){
+              	stop("null.model must have score.se.ratio when approx.score.SE = TRUE")
+              } 
+              if(calc.score && !(approx.score.SE) && isNullModelSmall(null.model)){
+              	stop("small null.model cannot be used when approx.score.SE = FALSE")
+              }
+              if(!is.null(GxE) && isNullModelSmall(nullmod)){
+              	stop("small null.model cannot be used with GxE")
+              }
+
```

Comment:
See above comment about an internal function - since you run the checks in multiple places, it'll be easier to maintain the code longterm if you call an internal function for them.